File: Examples_VI_CtCI_Python/Example10.py
import math

def prime(n):
    x = 2
    while (x <= math.sqrt(n)):
        if (n % x == 0):
            print(str(n) + " is not a prime number")
            return False
        x += 1
    print(str(n) + " is a prime number")
    return True

n = 4
#n = 5
prime(n)

File: Examples_VI_CtCI_Java/src/Example10.java

public class Example10 {
	//0 (sqrt n) runtime.
	
	static boolean isPrime(int n) {
		for (int x = 2; x * x <= n; x++) {
			if (n % x == 0) {
				System.out.println(n + " is not a prime number.");
				return false;
			}
		}
		System.out.println(n + " is a prime number.");
		return true;
	}
	
	public static void main(String[] args) {
		isPrime(10);
	}
}
